pandas/core/groupby/grouper.py

```diff
@@ -338,7 +344,14 @@ def _set_grouper(self, obj: FrameOrSeries, sort: bool = False):
             if getattr(self.grouper, "name", None) == key and isinstance(
                 obj, ABCSeries
             ):
-                ax = self._grouper.take(obj.index)
+                if group_indices is None:
```

Comment:
you can simply access `self._grouper.indices` no? why all of the convulated logic.

Reply:
Self._grouper is and DatetimeIndex in this case. It does not have the attribute indices unfortunately. The information about the groups are all lost.

Reply:
I am not that happy about the convolution too. But I wanted to avoid making group_indices required, so we have to catch the None case. Also if the index `self._grouper` was not resorted by the resample method, the indexer is None.

Reply:
something is not correct here, you need to really try and trace this; .indices a method on the grouper itself which *always* exists (there are different flavors, BinGroup and so on); its possible its not being passed thru correct. We need to fix at the correct level and avoid convolution.

Reply:
@jreback I think I got the issue now.

We have to different grouper parent classes. The class BaseGrouper from https://github.com/pandas-dev/pandas/blob/master/pandas/core/groupby/ops.py which **has** the indices method and the different parent class Grouper from

https://github.com/pandas-dev/pandas/blob/master/pandas/core/groupby/grouper.py

which **has no** method indices. The Groupings class from the same file has the method indices, but at the function above we have no Groupings object. Should I implement a indices method for the Grouper class?

Reply:
I am sorry for the confusion, but currently I can not see a way how to implement a indices method for a Grouper. The Grouper class does not hold the necessary information. When we want to get rid of the convolution, we will have to pass the indices object from the underlying BaseGrouper into the function. This seems pretty ugly, because we only need the indices in the Resample case, which is currently implemented and we would have to pass the indices object through a lot of layers to get it into this function at different places.

Reply:
maybe i don't understand what the actual problem is, but `.indices` is available on all groupers, (see pandas/core/groupby/grouper.py and search for `def indices`.)

groupby.resample is just a case of using the resampler grouper, which is generally a BinGrouper. 

I would encourage you to step into this line-by-line and really follow the functions (this is not trivial but provides lots of insight).

Reply:
Thanks for the answer. Maybe I miss something, then I would like to apologize in advance. But the way I understand this is the following:

In this specific resample case, we get a TimeGrouper, which inherits from Grouper.
Grouper does not have a indices Method. (https://github.com/pandas-dev/pandas/blob/e67220dcf98894bfe617efc3a3c879a0de7438bf/pandas/core/groupby/grouper.py#L35)

This file does have a indices method, but this method belongs to the Groupings class (
Class definition: https://github.com/pandas-dev/pandas/blob/e67220dcf98894bfe617efc3a3c879a0de7438bf/pandas/core/groupby/grouper.py#L388
indices definition: https://github.com/pandas-dev/pandas/blob/e67220dcf98894bfe617efc3a3c879a0de7438bf/pandas/core/groupby/grouper.py#L554
)

Groupings and Grouper have no relation with each other, so this does not help in this case.

We also have the BaseGrouper class, but this class also has no relation with Grouper or TimeGrouper.

I would really aprreciate it, if you could point out, where I am wrong. I stepped through the code, but the SeriesGroupBy class information with the indices method attached is lost here: 
https://github.com/pandas-dev/pandas/blob/1f49b76d5b07099c93af1780db73a0bfd74dbcc1/pandas/core/resample.py#L986

It was stored under self._groupby, but this information is not handed over